src/main/kotlin/repository/UserTableRepo.kt

```diff
@@ -21,6 +25,12 @@ interface IUserTableRepo {
     suspend fun getUserByEmail(email: String): User
 
     suspend fun getAllUsers(): List<User>
+
+    /**
+     * Returns 1 if the user was successfully soft deleted by the given [id], 0 otherwise.
+     * A soft delete means the user is not removed from the database, but their status is set to USER_STATUS_DELETED.
```

Comment:
Use `[]` for types. This way, it is ensured that they are valid.
```suggestion
     * A soft delete means the user is not removed from the database, but their status is set to [UserStatus.USER_STATUS_DELETED].
```